tools/Google.Cloud.Tools.GenerateSBOM/Program.cs

```diff
@@ -1,329 +1,329 @@
-        {
-            Namespaces = false
-        };
-        doc.Load(xmlReader);
-        var metadata = doc.DocumentElement?
-            .SelectSingleNode("/package/metadata");
-        return new NuspecMetadata
-        {
-            Id = metadata?.SelectSingleNode("id")?.InnerText ?? "",
-            Version = metadata?.SelectSingleNode("version")?.InnerText ?? "",
-            Description = metadata?.SelectSingleNode("description")?.InnerText
-                ?? "",
-        };
-    }
-
-    /// <summary>
-    /// Creates a UUID from a hash.
-    /// 
-    /// Ported from format_uuid_v3or5() in
-    /// https://www.rfc-editor.org/rfc/rfc4122
-    /// </summary>
-    /// <param name="hash">A SHA1 hash or the like.
-    ///   Must be 16 bytes or longer.</param>
-    /// <returns>A UUID deterministically created from the hash.</returns>
-    private static Guid Uuid5FromHash(byte[] hash)
-    {
-        /* convert UUID to local byte order */
-        var time_low = IPAddress.NetworkToHostOrder(
-            BitConverter.ToInt32(hash, 0));
-        var time_mid = IPAddress.NetworkToHostOrder(
-            BitConverter.ToInt16(hash, 4));
-        var time_hi_and_version = IPAddress.NetworkToHostOrder(
-            BitConverter.ToInt16(hash, 6));
-        var clock_seq_hi_and_reserved = hash[8];
-
-        /* put in the variant and version bits */
-        const short version = 5;
-        time_hi_and_version &= 0x0FFF;
-        time_hi_and_version |= version << 12;
-        clock_seq_hi_and_reserved &= 0x3F;
-        clock_seq_hi_and_reserved |= 0x80;
-        hash[8] = clock_seq_hi_and_reserved;
-
-        return new Guid(time_low, time_mid,
-            time_hi_and_version, hash[8..16]);
-    }
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License").
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at 
+//
+// https://www.apache.org/licenses/LICENSE-2.0 
+//
+// Unless required by applicable law or agreed to in writing, software 
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and 
+// limitations under the License.
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Net;
+using System.Security.Cryptography;
+using System.Text.Json;
+using System.Xml;
+
+namespace GenerateSBOM;
+
+// The classes below are not general purpose Spdx classes.  Their single
+// purpose is to generate json SBOMs for nupkgs generated from this repository,
+// so they have default values convenient for that purpose.
+// Reusing these classes for any other purpose would require a refactor.
+
+/// <summary>
+/// SPDX format.
+/// See https://github.com/spdx/spdx-spec/blob/development/v2.3.1/schemas/spdx-schema.json
+/// </summary>
+internal class Spdx
```

Comment:
There's quite a lot I'd want to change here in terms of C# conventions etc, but I think that's going to be best done as a follow-up PR after getting the initial logic in. Basically, I'll avoid nits in this review.

Reply:
Ok.  Thanks.